### DISCUSSION TITLE
Support for stylistic variant

### DISCUSSION THREAD

**Discussion by Xintong-Li** [olsak/OpTeX#79] [Ideas]
Some math fonts provide alternative stylistic set for users. For instance, by enabling stylistic set 2, users of New Computer Modern Math will get upright integrals. Similarly, by enabling stylistic set 8, STIX2 Math will enable upright integrals.

In unicode-math, this can be declared when setting up the font via \setmathfont{NewCMMath-Regular.otf}[StylisticSet=2] .

Is there any way to activate stylistic variant in OpTeX, or is there any further plan of supporting this?

Thank you.

**Comment by olsak** [comment]
You can use `otfinfo -f` to list all font features provided by given font. For example
```
otfinfo -f NewCMMath-Regular.otf
ss01	Stylistic Set 1
ss02	Stylistic Set 2
```
This means that this font supports two font features listed here. You can add directly the font features to the math font loaded by the `\lodmath` command (note that `\fontfam` uses it) by adding features to the internal `\_mfontfeatures` macro. So:
```
\addto\_mfontfeatures{ss02;}
\fontfam[newcm]

test: $\int f$
\bye
```
prints the math symbols from stylistic set 02.

Note, that you need not any conversion of font features of the font (`ss02` in our examaple) to a different LaTeX terminology (`StylisticSet=2` in your example from LaTeX `unicode-math`). Use directly font features listed by `otfinfo`.

**Reply by Xintong-Li** [reply]
Thank you for the explanation.
